nix/nim-status/default.nix

```diff
@@ -0,0 +1,150 @@
+{ newScope
+, mkShell
+, writeScript
+, writeText
+, xcodeWrapper
+, pkgs
+, stdenv
+, lib }:
+let
+  callPackage = newScope {};
+  buildNimStatus = platform: arch: callPackage ./nim-status.nix {
+    inherit platform arch;
+  };
+
+  # Metadata common to all builds of status-go
+  meta = {
+    description = "The Status Go module that consumes go-ethereum.";
+    license = lib.licenses.mpl20;
+    platforms = with lib.platforms; linux ++ darwin;
+  };
+
+  # Source can be changed with a local override from config
+  source = callPackage ./status-go-source.nix { };
+  
+  buildStatusGo = platform: arch: callPackage ./status-go.nix {
+    inherit platform arch meta source;
+  };
+
+  buildAndroid = buildMap: name: stdenv.mkDerivation {
+    name = "${name}-android-builder";
+    buildInputs = [ pkgs.coreutils ];
+    builder = writeScript "${name}-android-builder.sh"
+    ''
+      export PATH=${pkgs.coreutils}/bin:$PATH
+      mkdir $out
+
+      ln -s ${buildMap.x86} $out/x86
+      ln -s ${buildMap.arm} $out/armeabi-v7a
+      ln -s ${buildMap.arm64} $out/arm64-v8a
+    '';
+  };
+
+  # Create a single multi-arch fat binary using lipo
+  # Create a single header for multiple target archs
+  # by utilizing C preprocessor conditionals
+  buildIos = buildMap: name:
+    let
+      headerIos = writeText "${name}.h" ''
+        #if TARGET_CPU_X86_64
+        ${builtins.readFile "${buildMap.x86}/${name}.h"}
+        #elif TARGET_CPU_ARM
+        ${builtins.readFile "${buildMap.arm}/${name}.h"}
+        #else
+        ${builtins.readFile "${buildMap.arm64}/${name}.h"}
+        #endif
+      '';
+    in stdenv.mkDerivation {
+      inherit xcodeWrapper;
+      buildInputs = [ pkgs.coreutils ];
+      name = "${name}-ios-builder";
+      builder = writeScript "${name}-ios-builder.sh"
+      ''
+        export PATH=${pkgs.coreutils}/bin:${xcodeWrapper}/bin:$PATH
+        mkdir $out
+
+        # lipo merges arch-specific binaries into one fat iOS binary
+        lipo -create ${buildMap.x86}/lib${name}.a \
+             ${buildMap.arm}/lib${name}.a \
+             ${buildMap.arm64}/lib${name}.a \
+             -output $out/lib${name}.a
+
+        cp ${headerIos} $out/${name}.h
+        ${if name=="nim_status" then "cp ${buildMap.arm64}/nimbase.h $out" else ""}
```

Comment:
What is this? Why?

Reply:
We need the `nimbase.h` for nim-status builds only, not for status-go.

Reply:
Is there any harm in including it anyway for `status-go`? Unnecessary logic should be avoided.

Reply:
We'd have to either add it to the output of `nim-status/status-go.nix`, or use something like `cp nimbase.h || true` to suppress the error message when it's missing.

Reply:
I see. That's fine then.